src/core/Song.cpp

```diff
@@ -115,6 +116,10 @@ Song::Song() :
 			this, SLOT(masterVolumeChanged()), Qt::DirectConnection );
 /*	connect( &m_masterPitchModel, SIGNAL(dataChanged()),
 			this, SLOT(masterPitchChanged()));*/
+			
+	QTimer *timer = new QTimer(this);
+	connect(timer, &QTimer::timeout, [this](){Song::checkUpdateLength();});
```

Comment:
What about

```cpp
connect(this, &Song::updateLength, this, &Song::recalcLength, Qt::QueuedConnection);
```

similar to how `update()` and `repaint()` works, instead of the timer?

Reply:
I'm pretty sure this would run almost identically to how things are without the PR.  I don't think this throws out any redundant calls to updateLength.